### ISSUE TITLE
[Bug] [TouchEffect] GetIsAvailable --> System.ArgumentNullException: Value cannot be null.

### ISSUE THREAD

**Issue by Huaba93** [xamarin/XamarinCommunityToolkit#1216]
### Description
If you put a frame inside another frame and apply TouchEffect to both, it causes a null reference exception and crashes the app. 

I have only seen this on android and with frames. But I haven't tried much else in that direction either. 

```
System.ArgumentNullException: Value cannot be null.
Parameter name: bindable
  at Xamarin.CommunityToolkit.Effects.TouchEffect.GetIsAvailable (Xamarin.Forms.BindableObject bindable) [0x00015] in <5265966be8d2407fb7a4c3b38c6441d7>:0 
  at Xamarin.CommunityToolkit.Effects.TouchEffect.get_IsAvailable () [0x00006] in <5265966be8d2407fb7a4c3b38c6441d7>:0 
  at Xamarin.CommunityToolkit.Effects.TouchEffect.OnLayoutChildAdded (System.Object sender, Xamarin.Forms.ElementEventArgs e) [0x0001f] in <5265966be8d2407fb7a4c3b38c6441d7>:0 
  at Xamarin.CommunityToolkit.Effects.TouchEffect.SetChildrenInputTransparent (System.Boolean value) [0x0004a] in <5265966be8d2407fb7a4c3b38c6441d7>:0 
  at Xamarin.CommunityToolkit.Effects.TouchEffect.set_Element (Xamarin.Forms.VisualElement value) [0x0003e] in <5265966be8d2407fb7a4c3b38c6441d7>:0 
  at Xamarin.CommunityToolkit.Android.Effects.PlatformTouchEffect.OnAttached () [0x0003b] in <5265966be8d2407fb7a4c3b38c6441d7>:0 
  at Xamarin.Forms.Effect.SendAttached () [0x00009] in D:\a\1\s\Xamarin.Forms.Core\Effect.cs:54 
  at Xamarin.Forms.RoutingEffect.SendAttached () [0x00000] in D:\a\1\s\Xamarin.Forms.Core\RoutingEffect.cs:29 
  at Xamarin.Forms.Element.AttachEffect (Xamarin.Forms.Effect effect) [0x00045] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:513 
  at Xamarin.Forms.Element.set_EffectControlProvider (Xamarin.Forms.IEffectControlProvider value) [0x0007c] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:238 
  at Xamarin.Forms.Internals.EffectUtilities.RegisterEffectControlProvider (Xamarin.Forms.IEffectControlProvider self, Xamarin.Forms.IElementController oldElement, Xamarin.Forms.IElementController newElement) [0x0001a] in D:\a\1\s\Xamarin.Forms.Core\Internals\EffectUtilities.cs:16 
  at Xamarin.Forms.Platform.Android.FastRenderers.VisualElementRenderer.OnElementChanged (System.Object sender, Xamarin.Forms.Platform.Android.VisualElementChangedEventArgs e) [0x0005c] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\VisualElementRenderer.cs:102 
  at (wrapper delegate-invoke) System.EventHandler`1[Xamarin.Forms.Platform.Android.VisualElementChangedEventArgs].invoke_void_object_TEventArgs(object,Xamarin.Forms.Platform.Android.VisualElementChangedEventArgs)
  at Xamarin.Forms.Platform.Android.FastRenderers.FrameRenderer.OnElementChanged (Xamarin.Forms.Platform.Android.ElementChangedEventArgs`1[TElement] e) [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\FrameRenderer.cs:169 
  at Xamarin.Forms.Platform.Android.FastRenderers.FrameRenderer.set_Element (Xamarin.Forms.Frame value) [0x00018] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\FrameRenderer.cs:63 
  at Xamarin.Forms.Platform.Android.FastRenderers.FrameRenderer.Xamarin.Forms.Platform.Android.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00015] in D:\a\1\s\Xamarin.Forms.Platform.Android\FastRenderers\FrameRenderer.cs:84 
  at Xamarin.Forms.Platform.Android.Platform.CreateRenderer (Xamarin.Forms.VisualElement element, Android.Content.Context context) [0x0003e] in D:\a\1\s\Xamarin.Forms.Platform.Android\Platform.cs:358 
  at Xamarin.Forms.Platform.Android.VisualElementPackager.AddChild (Xamarin.Forms.VisualElement view, Xamarin.Forms.Platform.Android.IVisualElementRenderer oldRenderer, Xamarin.Forms.Platform.Android.RendererPool pool, System.Boolean sameChildren) [0x000b3] in D:\a\1\s\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:140 
  at Xamarin.Forms.Platform.Android.VisualElementPackager.SetElement (Xamarin.Forms.VisualElement oldElement, Xamarin.Forms.VisualElement newElement) [0x00163] in D:\a\1\s\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:351 
  at Xamarin.Forms.Platform.Android.VisualElementPackager.Load () [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.Android\VisualElementPackager.cs:112 
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetPackager (Xamarin.Forms.Platform.Android.VisualElementPackager packager) [0x00007] in D:\a\1\s\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:481 
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x000e8] in D:\a\1\s\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:244 
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].Xamarin.Forms.Platform.Android.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00033] in D:\a\1\s\Xamarin.Forms.Platform.Android\VisualElementRenderer.cs:131 
  at Xamarin.Forms.Platform.Android.Platform.CreateRenderer (Xamarin.Forms.VisualElement element, Android.Content.Context context) [0x0003e] in D:\a\1\s\Xamarin.Forms.Platform.Android\Platform.cs:358 
  at Xamarin.Forms.Platform.Android.AppCompat.Platform.AddChild (Xamarin.Forms.Page page, System.Boolean layout) [0x0000d] in D:\a\1\s\Xamarin.Forms.Platform.Android\AppCompat\Platform.cs:383 
  at Xamarin.Forms.Platform.Android.AppCompat.Platform.SetPageInternal (Xamarin.Forms.Page newRoot) [0x0005a] in D:\a\1\s\Xamarin.Forms.Platform.Android\AppCompat\Platform.cs:341 
  at Xamarin.Forms.Platform.Android.AppCompat.Platform.SetPage (Xamarin.Forms.Page newRoot) [0x000e6] in D:\a\1\s\Xamarin.Forms.Platform.Android\AppCompat\Platform.cs:295 
  at Xamarin.Forms.Platform.Android.FormsAppCompatActivity.InternalSetPage (Xamarin.Forms.Page page) [0x00039] in D:\a\1\s\Xamarin.Forms.Platform.Android\AppCompat\FormsAppCompatActivity.cs:507 
  at Xamarin.Forms.Platform.Android.FormsAppCompatActivity.SetMainPage () [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.Android\AppCompat\FormsAppCompatActivity.cs:533 
  at Xamarin.Forms.Platform.Android.FormsAppCompatActivity.LoadApplication (Xamarin.Forms.Application application) [0x00140] in D:\a\1\s\Xamarin.Forms.Platform.Android\AppCompat\FormsAppCompatActivity.cs:193 
  at TouchEffectNullreferenceNestedFrames.Android.MainActivity.OnCreate (Android.OS.Bundle savedInstanceState) [0x00027] in /Users/christophhuber/git/XamarinIssues/TouchEffectNullreferenceNestedFrames/TouchEffectNullreferenceNestedFrames/TouchEffectNullreferenceNestedFrames.Android/MainActivity.cs:22 
  at Android.App.Activity.n_OnCreate_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_savedInstanceState) [0x00010] in /Users/builder/azdo/_work/278/s/xamarin-android/src/Mono.Android/obj/Release/monoandroid10/android-30/mcw/Android.App.Activity.cs:2700 
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.5(intptr,intptr,intptr)
```

### Steps to Reproduce

```
    <Frame HorizontalOptions="Fill" Padding="50" BackgroundColor="LightGray"
           effects:TouchEffect.NativeAnimation="True"
           effects:TouchEffect.NativeAnimationShadowRadius="80"
           effects:TouchEffect.NativeAnimationColor="Red">
        <Frame HorizontalOptions="Fill" Padding="24" BackgroundColor="LightGray"
               effects:TouchEffect.NativeAnimation="True"
               effects:TouchEffect.NativeAnimationShadowRadius="80"
               effects:TouchEffect.NativeAnimationColor="Red" />
    </Frame>
```

Or see attached repo

### Basic Information

- Version with issue: 1.1.0
- Last known good version: -
- IDE:
- Platform Target Frameworks: 
  - iOS:  14.2
  - Android: API 30
- Nuget Packages:
- Affected Devices: Galaxy S10

### Reproduction Link
https://github.com/Huaba93/XamarinIssues/tree/master/TouchEffectNullreferenceNestedFrames

**Comment by Huaba93** [comment]
@AndreiMisiukevich I hope you could help with this.

**Comment by Cfun1** [comment]
I believe adding ` effects:TouchEffect.ShouldMakeChildrenInputTransparent="False"` to the first Frame won't crash the app and avoid this issue:
```
    <Frame HorizontalOptions="Fill" Padding="50" BackgroundColor="LightGray"
           effects:TouchEffect.NativeAnimation="True"
           effects:TouchEffect.ShouldMakeChildrenInputTransparent="False"
           effects:TouchEffect.NativeAnimationShadowRadius="80"
           effects:TouchEffect.NativeAnimationColor="Red">
        <Frame HorizontalOptions="Fill" Padding="24" BackgroundColor="LightGray"
               effects:TouchEffect.NativeAnimation="True"
               effects:TouchEffect.NativeAnimationShadowRadius="80"
               effects:TouchEffect.NativeAnimationColor="Red" />
    </Frame>
```

more context https://github.com/xamarin/XamarinCommunityToolkit/discussions/1146 https://github.com/xamarin/XamarinCommunityToolkit/issues/852

**Comment by AndreiMisiukevich** [comment]
> I believe adding ` effects:TouchEffect.ShouldMakeChildrenInputTransparent="False"` to the parent Frame will avoid this Android only issue:
> 
> ```
>     <Frame HorizontalOptions="Fill" Padding="50" BackgroundColor="LightGray"
>            effects:TouchEffect.NativeAnimation="True"
>            effects:TouchEffect.ShouldMakeChildrenInputTransparent="False"
>            effects:TouchEffect.NativeAnimationShadowRadius="80"
>            effects:TouchEffect.NativeAnimationColor="Red">
>         <Frame HorizontalOptions="Fill" Padding="24" BackgroundColor="LightGray"
>                effects:TouchEffect.NativeAnimation="True"
>                effects:TouchEffect.NativeAnimationShadowRadius="80"
>                effects:TouchEffect.NativeAnimationColor="Red" />
>     </Frame>
> ```
> 
> more context #1146 #852

Hey @Huaba93 could you please try to set ShouldMakeChildrenInputTransparent to false as proposed by @Cfun1 


P.S. Thanks both for feedback and help :)

**Comment by Huaba93** [comment]
Thanks for your help, it works now. But couldn't we make it failsafe somehow?

Is always setting `xct:TouchEffect.ShouldMakeChildrenInputTransparent="False"` only error prone for custom animations or for the native animation as well? Because on android I do not have any issue like #852 when using the native ripple effect.

**Comment by AndreiMisiukevich** [comment]
I believe you don't see any issues because your touchable views are not inside of ScrollView.
As for preventing crash -- totally agree. Need to fix that.

**Comment by Huaba93** [comment]
Sure I have tested it inside
* CarouselViews
* SwipeViews
* CollectionViews
* ScrollViews

No Problem at all on Android. But some on iOS. I hope I will soon find the time to create for each one an issue. I have rolled ou toucheffect for all our tapgesturerecognizers. That was a few hundred.
Unfortunately, the ripple effect noticeably affects the performance. This is probably due to additionally added elements. Is there anything we can do to avoid this? Probably only if we switch to scale animation or something similar. 
That was a few hundred. 

back to the topic: I still haven't figured out yet when exactly the null reference exception occurs. It is definitely view nesting related, but it occurs only rarely.
